### PR TITLE
MGMT-2252: Don't log secrets from the ignitionConfig

### PR DESCRIPTION
**Pull request by asalkeld**
This generates a new ignitionConfig without the secrets and logs it.

### CODE REVIEW

**Comment by gamli75** [comment]
@rollandf please review

**Review by ronniel1 — APPROVED** [review]
/lgtm
/Approve

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -279,6 +279,11 @@ func (b *bareMetalInventory) formatIgnitionFile(cluster *common.Cluster, params
 		"SkipCertVerification": strconv.FormatBool(b.SkipCertVerification),
 		"AgentTimeoutStartSec": strconv.FormatInt(int64(b.AgentTimeoutStart.Seconds()), 10),
 	}
+	if safeForLogs {
+		for _, key := range []string{"userSshKey", "PullSecretToken", "PULL_SECRET", "RH_ROOT_CA"} {
```
Comment (mhrivnak) [inline internal/bminventory/inventory.go:283]: Does the public part of a CA or the public part of an ssh key need to be considered secret?

Comment (asalkeld) [reply internal/bminventory/inventory.go:283]: well I guess strictly not, but I also thought what is the benefit of printing out a huge CA/ssh key?
I am happy to take it out if you think it is useful to have in.

**Pull request merged**